modules/launcher/src/main/java/org/mule/runtime/module/launcher/application/DefaultApplicationFactory.java

```diff
@@ -129,28 +145,45 @@ private ArtifactClassLoader createCompositePluginClassLoader(ArtifactClassLoader
         return new CompositeArtifactClassLoader("appPlugins", parent.getClassLoader(), classLoaders, parent.getClassLoaderLookupPolicy());
     }
 
-    private List<ApplicationPlugin> createApplicationPlugins(ArtifactClassLoader parentClassLoader, Set<ApplicationPluginDescriptor> pluginDescriptors)
+    private List<ApplicationPlugin> createContainerApplicationPlugins(ArtifactClassLoader parentClassLoader, ApplicationDescriptor appDescriptor) throws IOException
     {
         final List<ApplicationPlugin> plugins = new LinkedList<>();
 
-        for (ApplicationPluginDescriptor descriptor : pluginDescriptors)
+        File[] containerPlugins = MuleFoldersUtil.getContainerAppPluginsFolder().listFiles();
+        if (containerPlugins != null)
         {
-            final MuleArtifactClassLoader pluginClassLoader = createPluginClassLoader(parentClassLoader, descriptor);
-            final DefaultApplicationPlugin applicationPlugin = new DefaultApplicationPlugin(descriptor, pluginClassLoader);
-
-            plugins.add(applicationPlugin);
+            for (File pluginFolder : MuleFoldersUtil.getContainerAppPluginsFolder().listFiles((FileFilter) new SuffixFileFilter(".zip", IOCase.INSENSITIVE)))
```

Comment:
Descriptor for plugins contained in the container would not change. Makes sense to maintain the list in some place instead of recreating it on each deploy